server/src/test/java/org/elasticsearch/search/vectors/VectorDataTests.java

```diff
@@ -0,0 +1,172 @@
+        XContentParser parserMock = mock(XContentParser.class);
+        when(parserMock.currentToken()).thenReturn(XContentParser.Token.START_ARRAY);
+        when(parserMock.nextToken()).thenReturn(
+            XContentParser.Token.VALUE_NUMBER,
+            XContentParser.Token.VALUE_NUMBER,
+            XContentParser.Token.VALUE_NUMBER,
+            XContentParser.Token.VALUE_NUMBER,
+            XContentParser.Token.END_ARRAY
+        );
+        when(parserMock.floatValue()).thenReturn(64f, 10f, -30f, 10f);
+        VectorData parsed = VectorData.parseXContent(parserMock);
+        assertArrayEquals(expectedByteArray, parsed.asByteVector());
+    }
+
+    public void testByteThrowsForOutsideRange() throws IOException {
+        XContentParser parserMock = mock(XContentParser.class);
+        when(parserMock.currentToken()).thenReturn(XContentParser.Token.START_ARRAY);
+        when(parserMock.nextToken()).thenReturn(XContentParser.Token.VALUE_NUMBER, XContentParser.Token.END_ARRAY);
+        when(parserMock.floatValue()).thenReturn(1000f);
+        VectorData parsed = VectorData.parseXContent(parserMock);
+        IllegalArgumentException ex = expectThrows(IllegalArgumentException.class, parsed::asByteVector);
+        assertThat(ex.getMessage(), containsString("vectors only support integers between [-128, 127]"));
+    }
+
+    public void testAsByteThrowsForDecimals() throws IOException {
+        XContentParser parserMock = mock(XContentParser.class);
+        when(parserMock.currentToken()).thenReturn(XContentParser.Token.START_ARRAY);
+        when(parserMock.nextToken()).thenReturn(XContentParser.Token.VALUE_NUMBER, XContentParser.Token.END_ARRAY);
+        when(parserMock.floatValue()).thenReturn(0.1f);
+        VectorData parsed = VectorData.parseXContent(parserMock);
+        IllegalArgumentException ex = expectThrows(IllegalArgumentException.class, parsed::asByteVector);
+        assertThat(ex.getMessage(), containsString("vectors only support non-decimal values but found decimal value"));
+    }
+
+    public void testParseSingleNumber() throws IOException {
+        VectorData expected = new VectorData(new float[] { 0.1f }, null);
+        XContentParser parserMock = mock(XContentParser.class);
+        when(parserMock.currentToken()).thenReturn(XContentParser.Token.VALUE_NUMBER);
+        when(parserMock.floatValue()).thenReturn(0.1f);
+        VectorData parsed = VectorData.parseXContent(parserMock);
+        assertEquals(expected, parsed);
+    }
+
+    public void testParseThrowsForUnknown() {
+        XContentParser parserMock = mock(XContentParser.class);
+        when(parserMock.currentToken()).thenReturn(XContentParser.Token.START_OBJECT);
+        ParsingException ex = expectThrows(ParsingException.class, () -> VectorData.parseXContent(parserMock));
+        assertThat(ex.getMessage(), containsString("Unknown type [" + XContentParser.Token.START_OBJECT + "] for parsing vector"));
+    }
```

Comment:
```suggestion
    public void testParseThrowsForUnknown() throws IOException {
        String unknown = "{\"foo\":\"bar\"}";
        try(XContentParser parser = XContentHelper.createParser(
            XContentParserConfiguration.EMPTY,
            new BytesArray(unknown),
            XContentType.JSON)) {
            parser.nextToken();
            ParsingException ex = expectThrows(ParsingException.class, () -> VectorData.parseXContent(parser));
            assertThat(ex.getMessage(), containsString("Unknown type [" + XContentParser.Token.START_OBJECT + "] for parsing vector"));
        };
    }
```

I would much prefer us not use mocks. Here is an example of how to do this particular test without mocks.

Reply:
Thanks for the suggestion - will update accordingly :)

Reply:
Addressed in 2c346822d6d66feab114538f4dd9e2b1313251f0